utils/parametric/_library_client.py

```diff
@@ -941,41 +959,50 @@ def otel_start_span(
             "trace_id": resp["trace_id"],
         }
 
-    def current_span(self) -> Union[_TestSpan, None]:
-        resp = self._client.current_span()
-        if resp is None:
-            return None
-        return _TestSpan(self._client, resp["span_id"], resp["trace_id"])
-
-    def flush(self):
-        self._client.trace_flush()
+    def flush(self) -> bool:
+        return self._client.trace_flush()
 
     def otel_flush(self, timeout_sec: int) -> bool:
         return self._client.otel_flush(timeout_sec)
 
-    def otel_current_span(self) -> Union[_TestOtelSpan, None]:
-        resp = self._client.otel_current_span()
-        if resp is None:
-            return None
-        return _TestOtelSpan(self._client, resp["span_id"], resp["trace_id"])
-
     def otel_is_recording(self, span_id: int) -> bool:
         return self._client.otel_is_recording(span_id)
 
     def inject_headers(self, span_id) -> List[Tuple[str, str]]:
         return self._client.trace_inject_headers(span_id)
 
     def otel_set_baggage(self, span_id: int, key: str, value: str):
-        return self._client.otel_set_bagage(span_id, key, value)
+        return self._client.otel_set_baggage(span_id, key, value)
+
+    def finish_span(self, span_id: int) -> None:
+        self._client.finish_span(span_id)
+
+    def get_tracer_config(self) -> Dict[str, Optional[str]]:
+        return self._client.get_tracer_config()
+
+    ### Do not use the methods below in parametric tests, they will be removed in a future PR ####
 
     def http_client_request(
         self, url: str, method: str = "GET", headers: List[Tuple[str, str]] = None, body: Optional[bytes] = b"",
     ):
         """Do an HTTP request with the given method and headers."""
         return self._client.http_client_request(method=method, url=url, headers=headers or [], body=body,)
 
-    def finish_span(self, span_id: int) -> None:
-        self._client.finish_span(span_id)
+    def current_span(self) -> Union[_TestSpan, None]:
+        resp = self._client.current_span()
+        if resp is None:
+            return None
+        return _TestSpan(self._client, resp["span_id"], resp["trace_id"])
 
-    def get_tracer_config(self) -> Dict[str, Optional[str]]:
-        return self._client.get_tracer_config()
+    def otel_current_span(self) -> Union[_TestOtelSpan, None]:
+        resp = self._client.otel_current_span()
+        if resp is None:
+            return None
+        return _TestOtelSpan(self._client, resp["span_id"], resp["trace_id"])
+
+    def is_alive(self) -> bool:
```

Comment:
How will this be used in future PRs?

Reply:
It will be used in crash tracking tests or any test that should kill the parametric app.

Reply:
So I know that the .NET parametric app crashes correctly, which leads me to think that this may be flaky. It looks like the `self._client.wait` command calls `container.reload()` which seems wrong. Can you instead do a check on the container object instead of calling wait?

Reply:
Ah that makes sense. The python app fails to reload. I will clean this up